STM32H7RS: DCMIPP: enabling of camera pipe with ST-B-CAMS-OMV shield

avolmat-st opened the pull request:
This PR enables the DCMIPP based camera pipeline on the STM32H7RS. This is being tested by usage of the ST-B-CAMS-OMV shield (DVP) on the STM32H7S78-DK.

Tested ok via:
```
west build -p -b stm32h7s78_dk//ext_flash_app --shield st_b_cams_omv_mb1683 samples/drivers/video/capture --sysbuild
```

Commit:
dts: arm: st: add dcmipp node in stm32h7rs.dtsi

The STM32H7RS series embeds a parallel interface based DCMIPP
block allowing to capture data from sensors and store them into
memory.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
boards: st: add camera connector and camera interface in stm32h7s78

Describe the camera related pins of the 30pins connectors of the
stm32h7s78 as well as DCMIPP pin assignments for the stm32h7s78

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
boards: st: stm32h7s78_dk: use PSRAM for video buffers

Use the psram to store video buffers.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
drivers: clock: h7rs: add PCLKx support

Addition of PCLKx clock source which can be used by some
peripherals.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

Commit:
boards: shields: st_b_cams_omv_mb1683: add stm32h7s78_dk confs

Add board specific configuration and overlay for enabling the
st_b_cams_omv_mb1683 shield on the stm32h7s78_dk board.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

avolmat-st commented on dts/arm/st/h7rs/stm32h7rs.dtsi:
```diff
@@ -1003,6 +1004,26 @@
 		};
+
+		dcmipp: dcmipp@50002000 {
+			compatible = "st,stm32h7rs-dcmipp", "st,stm32-dcmipp";
```
Actually no it isn't really needed now. I actually did that for other platforms as well N6 / MP13 but only st,stm32-dcmipp compatible is always used.
In Linux world it is common to have a soc specific compatible and a generic one but maybe here it isn't really needed in fact and we are in fact relying on other stuff to see if this is MP13 / N6 etc.
Shall I remove those soc specific compatible for N6 / MP13 and do not put it for H7RS ?

avolmat-st replied:
Oups, my bad. Removed.

Commit:
video: st: remove soc specific dcmipp compatibles

Only st,stm32-dcmipp is described and only st,stm32n6-dcmipp
was used within the driver to decide if CSI / PIXEL_PIPES
are available. Instead of this, look at HAL provided macros
to know if the selected soc has the functionalities or not.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

ngphibang reviewed (dismissed):
LGTM for the video side.

etienne-lms reviewed (changes requested):
Please cross-check the PE4 pin muxing review comment.

avolmat-st commented on boards/st/stm32h7s78_dk/stm32h7s78_dk-common.dtsi:
```diff
@@ -353,3 +375,11 @@ zephyr_udc0: &usb2 {};
+
+st_cam_dvp: &dcmipp {
+	pinctrl-0 = <&dcmipp_d0_pc6 &dcmipp_d1_pc7 &dcmipp_d2_pe0 &dcmipp_d3_pe1
+		     &dcmipp_d4_pe4 &dcmipp_d5_pd3 &dcmipp_d6_pb8 &dcmipp_d7_pd14
```
Good point. Actually I think the SPI4 entry is not correct.
The board manual (https://www.st.com/resource/en/user_manual/um3289-discovery-kit-with-stm32h7s7l8-mcu-stmicroelectronics.pdf) indicates PE10, however such muxing is not a valid out of the SPI4 instance
(https://www.st.com/resource/en/datasheet/stm32h7s3a8.pdf)

I think that the SPI CS should actually be controlled out of PE10, as a GPIO.
I had a commit into the PR to fix this prior to adding the dcmipp pinctrl entry.

Commit:
boards: st: stm32h7s78_dk: correct SPI4 chip select

On this board the chip select is connected to GPIO E10
and must be controlled via GPIO since SPI4 CS is accessible
via either E4 or E11 which are both already used by other
functions on this board.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

etienne-lms reviewed (approved):
LGTM, AFAICT.
